Asset Uploader UI Component/Code/Asset Uploader/Objective-C/AssetUploader/ViewController.m

```diff
@@ -261,7 +278,44 @@ - (void)syncFinished
     [[AdobeLibraryManager sharedInstance] deregisterDelegate:self];
 }
 
-#pragma mark - Priavte methods
+#pragma mark - Notification handlers
+
+- (void)authDidLogout:(NSNotification *)notification
+{
+    if (self.rootLibDir.length > 0)
+    {
+        [AdobeLibraryManager removeLocalLibraryFilesInRootFolder:self.rootLibDir withError:nil];
+        
+        // Clean up the root folder.
+        NSFileManager *fm = [NSFileManager defaultManager];
+        [fm removeItemAtPath:[[self.rootLibDir stringByDeletingLastPathComponent] stringByDeletingLastPathComponent] error:nil];
+        self.rootLibDir = nil;
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:@"userSpecificPathKey"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+}
+
+#pragma mark - Private methods
+
+- (NSString *)UUIDForUserSpecificPath
+{
+    // If we have a UUID then use it.
+    [[NSUserDefaults standardUserDefaults] synchronize];
+    NSString *userSpecificID = [[NSUserDefaults standardUserDefaults] objectForKey:@"userSpecificPathKey"];
+    
+    if (userSpecificID.length > 0)
+    {
+        return userSpecificID;
+    }
+    
+    // Generate a new UUID
+    userSpecificID = [[NSUUID UUID] UUIDString];
+    
+    [[NSUserDefaults standardUserDefaults] setObject:userSpecificID forKey:@"userSpecificPathKey"];
+    [[NSUserDefaults standardUserDefaults] synchronize];
```

Comment:
We don’t need to call `synchronize` here. Per the documentation:

> Because this method is automatically invoked at periodic intervals, use this method only if you cannot wait for the automatic synchronization (for example, if your application is about to exit) or if you want to update the user defaults to what is on disk even though you have not made any changes.

Furthermore, since iOS 8, calling `synchronize` isn’t really necessary in most cases. The [Foundation Release Notes](
https://developer.apple.com/library/prerelease/content/releasenotes/Foundation/RN-Foundation-v10.10/index.html#10_10UserDefaults) for iOS 8 state:

> In earlier releases, the various ‘set’ methods on NSUserDefaults would wait several seconds before publishing the changes to other applications unless synchronized (though it has always been instant within a program). This delay has been removed.